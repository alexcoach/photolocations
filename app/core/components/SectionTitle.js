// @flow
import React from 'react';
import { StyleSheet } from 'react-native';
import { View, Text, Icon } from 'native-base';

import theme from 'app/theme';

type Props = {
    icon?: Object,
    text?: string,
    right?: any,
    style?: Object,
};

export default class SectionTitle extends React.PureComponent<Props> {
    renderRight = () => {
        const { right } = this.props;
        if (!right) {
            return null;
        }
        return (
            <View>
                {right}
            </View>
        );
    };
    renderIcon = () => {
        const { icon } = this.props;
        if (!icon) {
            return null;
        }
        return (
            <Icon {...icon} style={styles.icon} />
        );
    };
    render() {
        const { text, style } = this.props;
        return (
            <View style={[styles.row, styles.wrap, style]}>
                {this.renderIcon()}
                <View style={styles.flex}>
                    <Text>{text}</Text>
                </View>
                {this.renderRight()}
            </View>
        );
    }
}

const styles = StyleSheet.create({
    wrap: {
        width: '100%',
        backgroundColor: theme.muted,
        paddingHorizontal: 20,
        paddingVertical: 10,
    },
    row: {
        flexDirection: 'row',
        alignItems: 'center',
    },
    flex: {
        flex: 1,
    },
    icon: {
        marginRight: 5,
    },
});
