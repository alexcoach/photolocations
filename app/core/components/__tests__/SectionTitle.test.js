import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import { Text } from 'native-base';
import SectionTitle from '../SectionTitle';

it('<SectionTitle> renders correctly', () => {
    const wrapper = shallow(
        <SectionTitle
            icon={{ name: 'home' }}
            text="Test Title"
            right={<Text>right</Text>}
            style={{ marginTop: 10 }}
        />
    );
    expect(wrapper.dive()).toMatchSnapshot();
});
