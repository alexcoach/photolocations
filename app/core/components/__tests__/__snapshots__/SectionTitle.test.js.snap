// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionTitle> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Styled(ViewNB)
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        Object {
          "backgroundColor": "#dddddd",
          "paddingHorizontal": 20,
          "paddingVertical": 10,
          "width": "100%",
        },
        Object {
          "marginTop": 10,
        },
      ]
    }
  >
    <Styled(Icon)
      name="home"
      style={
        Object {
          "marginRight": 5,
        }
      }
    />
    <Styled(ViewNB)
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Styled(Text)>
        Test Title
      </Styled(Text)>
    </Styled(ViewNB)>
    <Styled(ViewNB)>
      <Styled(Text)>
        right
      </Styled(Text)>
    </Styled(ViewNB)>
  </Styled(ViewNB)>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(Icon)
          name="home"
          style={
            Object {
              "marginRight": 5,
            }
          }
        />,
        <Styled(ViewNB)
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Styled(Text)>
            Test Title
          </Styled(Text)>
        </Styled(ViewNB)>,
        <Styled(ViewNB)>
          <Styled(Text)>
            right
          </Styled(Text)>
        </Styled(ViewNB)>,
      ],
      "style": Array [
        Object {},
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        Object {
          "backgroundColor": "#dddddd",
          "paddingHorizontal": 20,
          "paddingVertical": 10,
          "width": "100%",
        },
        Object {
          "marginTop": 10,
        },
      ],
      "virtual": undefined,
    },
    "ref": [Function],
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "name": "home",
          "style": Object {
            "marginRight": 5,
          },
          "virtual": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Styled(Text)>
            Test Title
          </Styled(Text)>,
          "style": Object {
            "flex": 1,
          },
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Test Title",
            "virtual": undefined,
          },
          "ref": null,
          "rendered": "Test Title",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Styled(Text)>
            right
          </Styled(Text)>,
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "right",
            "virtual": undefined,
          },
          "ref": null,
          "rendered": "right",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(Icon)
            name="home"
            style={
              Object {
                "marginRight": 5,
              }
            }
          />,
          <Styled(ViewNB)
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Styled(Text)>
              Test Title
            </Styled(Text)>
          </Styled(ViewNB)>,
          <Styled(ViewNB)>
            <Styled(Text)>
              right
            </Styled(Text)>
          </Styled(ViewNB)>,
        ],
        "style": Array [
          Object {},
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
          Object {
            "backgroundColor": "#dddddd",
            "paddingHorizontal": 20,
            "paddingVertical": 10,
            "width": "100%",
          },
          Object {
            "marginTop": 10,
          },
        ],
        "virtual": undefined,
      },
      "ref": [Function],
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "name": "home",
            "style": Object {
              "marginRight": 5,
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(Text)>
              Test Title
            </Styled(Text)>,
            "style": Object {
              "flex": 1,
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Test Title",
              "virtual": undefined,
            },
            "ref": null,
            "rendered": "Test Title",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(Text)>
              right
            </Styled(Text)>,
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "right",
              "virtual": undefined,
            },
            "ref": null,
            "rendered": "right",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
